pkg/apis/controller/suggestions/v1alpha3/suggestion_types.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	common "github.com/kubeflow/katib/pkg/apis/controller/common/v1alpha3"
+)
+
+// SuggestionSpec defines the desired state of Suggestion
+type SuggestionSpec struct {
+	// Number of suggestions requested
+	Suggestions *int32 `json:"suggestions,omitempty"`
```

Comment:
Can this be nil?

Reply:
I think so.

Reply:
If it is not set, it will be nil

Reply:
Which situation can it get nil? I feel that Suggestion CR will always have non zero value.  Do you see any other scenario?

Reply:
I think it is for https://github.com/kubeflow/katib/pull/726#discussion_r320178130.

The defaulting webhook can set the field to 0 if it is nil.